include/RAJA/LegacyCompatibility.hxx

```diff
@@ -0,0 +1,215 @@
+#ifndef KRIPKE_LEGAGY_COMPATIBILITY_HXX
+#define KRIPKE_LEGAGY_COMPATIBILITY_HXX 1
+
+static_assert(__cplusplus >= 201103L, "C++ standards below 2011 are not supported");
+
+#include <type_traits>
+#include <utility>
+#include <iostream>
+#include <functional>
+#include <cstdint>
+
+namespace VarOps {
```

Comment:
Todd,

Without thinking too hard, I believe that would work.

—Rich

From: Todd Gamblin <notifications@github.com<mailto:notifications@github.com>>
Reply-To: LLNL/RAJA <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Friday, May 6, 2016 at 8:40 AM
To: LLNL/RAJA <RAJA@noreply.github.com<mailto:RAJA@noreply.github.com>>
Subject: Re: [LLNL/RAJA] Codegen must die (#7)


In include/RAJA/LegacyCompatibility.hxx<https://github.com/LLNL/RAJA/pull/7#discussion_r62347382>:

> @@ -0,0 +1,215 @@
> +#ifndef KRIPKE_LEGAGY_COMPATIBILITY_HXX
> +#define KRIPKE_LEGAGY_COMPATIBILITY_HXX 1
> +
> +static_assert(__cplusplus >= 201103L, "C++ standards below 2011 are not supported");
> +
> +#include <type_traits>
> +#include <utility>
> +#include <iostream>
> +#include <functional>
> +#include <cstdint>
> +
> +namespace VarOps {


On this note: John Mellor-Crummey was asking whether we could somehow tag the top-level entry points for RAJA so he can do better collapsing of nested/inlined calls in HPCToolkit. Basically he wants to show users a profile that only includes, e.g. RAJA:foraall but not all the plumbing underneath.

I think you could do this by putting all the app entry points in the RAJA namespace and moving internal calls out to other namespaces. Then tools would have nice labels and could pick what to show. Thoughts?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/LLNL/RAJA/pull/7/files/2e4346c272a4383ea030c6365f55a27ac4181f10#r62347382>